yapapi/executor/_smartq.py

```diff
@@ -126,7 +126,13 @@ async def has_unassigned_items(self) -> bool:
         A queue has unassigned items iff the next call to `get()` will immediately return
         some item, without waiting for an item that is currently "in progress" to be rescheduled.
         """
-        return await self.has_new_items() or bool(self._rescheduled_items)
+        while True:
+            if self._rescheduled_items:
+                return True
+            try:
+                return await asyncio.wait_for(self.has_new_items(), 1.0)
+            except asyncio.TimeoutError:
+                pass
```

Comment:
Instead of waiting indefinitely on potentialy blocking `await self.has_new_items()`, periodically check if there are `self._rescheduled_items`.